File: survey-builder-api/src/main/java/com/myapp/surveybuilderapi/entity/Survey.java
package com.myapp.surveybuilderapi.entity;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import java.time.Instant;
import java.util.List;
import java.util.Set;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@Builder
@Entity
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "survey")
public class Survey extends AbstractAuditEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;
    private String name;
    @Column(length = 2000)
    private String description;
    private boolean isAllowAnonymous;// TODO: now this field auto true
    private boolean isSpecificUser; // TODO: now this field auto false
    // if isSpecificUser is true, then survey only allow user in organization take a survey.
    private String organization;
    private Instant startDate;
    private Instant endDate;
    @Column(name = "is_del")
    private boolean isDel;

    @OneToMany(fetch = FetchType.LAZY, cascade = CascadeType.ALL, mappedBy = "survey")
    private Set<Question> questions;

    @OneToMany(fetch = FetchType.LAZY, cascade = CascadeType.ALL, mappedBy = "survey")
    private List<Answer> answers;
}

File: ui/src/types/Survey.ts
import { Question, QuestionData } from "./Question";

export type Survey = {
  id: string;
  name: string;
  description: string;
  startDate: string;
  endDate: string;
  remainingDate: number;
  count: number;
  isAllowAnonymous: boolean;
};

export type SurveyList = {
  surveys: Survey[];
  total: number;
};

export type SurveyCreate = {
  name: string;
  description: string;
  startDate?: string;
  endDate?: string;
  questions: Question[];
};

export type SurveyData = {
  id: string;
  name: string;
  description: string;
  questions: QuestionData[];
};
